lib/matplotlib/collections.py

```diff
@@ -1509,42 +1509,18 @@ def __init__(self,
         --------
         .. plot:: gallery/lines_bars_and_markers/eventcollection_demo.py
         """
-        if positions is None:
-            raise ValueError('positions must be an array-like object')
-        # Force a copy of positions
-        positions = np.array(positions, copy=True)
-        segment = (lineoffset + linelength / 2.,
-                   lineoffset - linelength / 2.)
-        if positions.size == 0:
-            segments = []
-        elif positions.ndim > 1:
-            raise ValueError('positions cannot be an array with more than '
-                             'one dimension.')
-        elif (orientation is None or orientation.lower() == 'none' or
-              orientation.lower() == 'horizontal'):
-            positions.sort()
-            segments = [[(coord1, coord2) for coord2 in segment] for
-                        coord1 in positions]
-            self._is_horizontal = True
-        elif orientation.lower() == 'vertical':
-            positions.sort()
-            segments = [[(coord2, coord1) for coord2 in segment] for
-                        coord1 in positions]
-            self._is_horizontal = False
-        else:
-            cbook._check_in_list(['horizontal', 'vertical'],
-                                 orientation=orientation)
-
         LineCollection.__init__(self,
-                                segments,
+                                [],
                                 linewidths=linewidth,
                                 colors=color,
                                 antialiaseds=antialiased,
                                 linestyles=linestyle,
                                 **kwargs)
-
+        self._is_horizontal = True  # Initial value, may be switched below.
```

Comment:
May this be just `self._is_horizontal = orientation` (and no need in `self.set_orientation(orientation)`)?

Reply:
no, set_orientation handles the checking of the argument as well (mapping "none" and "horizontal" case-insensitively to True, etc.)

Reply:
If it is the only reason to introduce segments flipping (even conditional) to constructor I think extracting `"none"`/`"horizontal"`/`"vertical"` to bool mapping into its own function is a better choice.

Reply:
I personally think it's cleaner to just have that in the setter.

Reply:
Since `segments` is set to an empty list above, you could just avoid the segment flipping by calling `set_orientation` before `set_positions`, couldn't you?

Reply:
sure.